### ISSUE TITLE
Modifying state during view update, this will cause undefined behavior.

### ISSUE THREAD

**Issue by Ceylo** [yumemi-inc/RepresentableKit#1]
Hello,

First of all I like the idea behind this library! However in practice it doesn't look usable yet:
<img width="1140" alt="image" src="https://user-images.githubusercontent.com/451334/147989906-697469e4-20b7-48d7-ac58-54dd89ea6fe3.png">

This is what I get in some toy project, however in a real use case this issue is actually hitting hard with a crash both in debug and release due to this precondition failure:
```
[error] precondition failure: setting value during update: 2072
```
As far as I see, all methods exposed by UIViewRepresentable are called during the view update, so no state can be changed there and the current approach seems to hit a dead end.

I'm using Xcode 13.2.1 on macOS 12.1.

**Comment by auramagi** [comment]
Thank you for opening this issue and sorry for taking so long to reply!
While making the library, I was focused on making previews work and the runtime warning just skipped my attention.  I took a look at it and figured a way to make it work, you can see the changes in this PR:
  - #2

**Comment by Ceylo** [comment]
I don't know for the changes in the PR, but at least when I try it I don't get the above issue anymore!
I'll be able to have 3 times less lines of code for a SwiftUI view thanks to your library 🙂

My use case is just a simple UITextView, as SwiftUI's Text doesn't support some features I need.
```swift
struct TextView: View {
    var text: AttributedString
    
    var body: some View {
        UIViewAdaptor {
            let view = UITextView()
            view.isEditable = false
            view.isScrollEnabled = false
            view.attributedText = NSAttributedString(text)
            return view
        }
    }
}
```

**Comment by auramagi** [comment]
Hey @Ceylo, glad to hear this resolves your issue and that you could find a use for the library.
I've merged the changes and released them in version 0.1.2.
